src/hiero_analytics/analysis/contributor_churn.py

```
@@ -0,0 +1,103 @@
+    progression = df.groupby("author").agg({
+        "level": list,
+        "pr_merged_at": ["min", "max"],
+        "pr_number": "nunique"
+    })
+    progression.columns = ["levels", "first_seen", "last_seen", "pr_count"]
+    
+    level_order = {spec.name: i for i, spec in enumerate(DIFFICULTY_LEVELS)}
+    level_order["Unknown"] = -1
+    
+    progression["max_level"] = progression["levels"].apply(
+        lambda lvls: max(lvls, key=lambda l: level_order.get(l, -1))
```

Ruff (0.15.10)

[error] Ambiguous variable name: `l`

(E741)